### PR TITLE
Set pipefail for root and root-virtio consoles

### PR DESCRIPTION
**Pull request by mloviska**
Bash option `pipefail` [is not set for all consoles](http://kepler.suse.cz/tests/1871#step/validate_packages_and_patterns/2) leading to false positive results as in below test job.
Issue occurs only in test jobs using virtio console.

- False positive: [jeos-extratest_kvm@64bit-virtio-vga ](https://openqa.suse.de/tests/4726367#step/validate_packages_and_patterns/10)
- Verification runs: 
  * [Build20.35-jeos-extratest_kvm@64bit-virtio-vga](http://kepler.suse.cz/tests/1929#step/validate_packages_and_patterns/13)
  * [sle-15-SP2-JeOS-for-kvm-and-xen-x86_64-Build15.39-jeos-extratest_kvm@64bit-virtio-vga; BETA=1](http://kepler.suse.cz/tests/1931#step/validate_packages_and_patterns/14)
   * [Build20.44-jeos-extratest_xenpv@svirt-xen-pv](http://kepler.suse.cz/tests/1927#step/validate_packages_and_patterns/13)
   * [Build20.44-jeos-extratest_xenhvm@svirt-xen-hvm](http://kepler.suse.cz/tests/1925#step/validate_packages_and_patterns/13)
   * [Build20.44-jeos-extratest_hyperv@svirt-hyperv](http://kepler.suse.cz/tests/1926#step/validate_packages_and_patterns/14)
   * [Build48.1-extra_tests_textmode@64bit](http://kepler.suse.cz/tests/1928#step/validate_packages_and_patterns/13)
   * [Build20200929-jeos@64bit_virtio](http://kepler.suse.cz/tests/1930#)
- [Missing upgrade packages](https://bugzilla.suse.com/show_bug.cgi?id=1177118)

### CODE REVIEW

File: tests/console/consoletest_setup.pm
Comment (jlausuch) [inline tests/console/consoletest_setup.pm:73]: my $res = script_run 'echo ${SHELLOPTS} | grep -w pipefail';
    script_run '. /etc/bash.bashrc.local' if $res;

Comment (jlausuch) [reply tests/console/consoletest_setup.pm:73]: I guess you want to source that rc file if the command fails (return code !=0).

**Comment by asmorodskyi** [comment]
@mloviska  I assume you have seen lines 68-69 of same file. Sounds like a bug that we need to run same things twice . Before merging this we need to figure out if it is product bug or test code one and maybe tune up your patch accordingly

File: tests/console/consoletest_setup.pm
Comment (Vogtinator) [reply tests/console/consoletest_setup.pm:73]: Isn't that what it already does?

Comment (asmorodskyi) [reply tests/console/consoletest_setup.pm:73]: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/2217

**Comment by Vogtinator** [comment]
> @mloviska I assume you have seen lines 68-69 of same file. Sounds like a bug that we need to run same things twice . Before merging this we need to figure out if it is product bug or test code one and maybe tune up your patch accordingly

Yeah, this PR seems like a noop. The issue seems to be that `script_run` runs in the virtio console, but some later tests use ttys which were logged into before `bashrc` was edited. So either `consoletest_setup` has to run before anything else use ttys or log out all logged in ttys.

**Comment by mloviska** [comment]
> 
> Yeah, this PR seems like a noop. The issue seems to be that `script_run` runs in the virtio console, but some later tests use ttys which were logged into before `bashrc` was edited. So either `consoletest_setup` has to run before anything else use ttys or log out all logged in ttys.

Yes, this is the case basically. Ok, let me try to shift `consoletest_setup` in the schedule

**Comment by Vogtinator** [comment]
> > Yeah, this PR seems like a noop. The issue seems to be that `script_run` runs in the virtio console, but some later tests use ttys which were logged into before `bashrc` was edited. So either `consoletest_setup` has to run before anything else use ttys or log out all logged in ttys.
> 
> Yes, this is the case basically. Ok, let me try to shift `consoletest_setup` in the schedule

In the JeOS case that's practically impossible - it would have to run before `firstrun`. So the only option is to either log out all consoles or run the command in those as well.

**Comment by mloviska** [comment]
> > > Yeah, this PR seems like a noop. The issue seems to be that `script_run` runs in the virtio console, but some later tests use ttys which were logged into before `bashrc` was edited. So either `consoletest_setup` has to run before anything else use ttys or log out all logged in ttys.
> > 
> > 
> > Yes, this is the case basically. Ok, let me try to shift `consoletest_setup` in the schedule
> 
> In the JeOS case that's practically impossible - it would have to run before `firstrun`. So the only option is to either log out all consoles or run the command in those as well.

I was thinking to set 'root-console' instead of virtio for `consoletest_setup` also. Thus, it would set pipefail in root-console and prepare bashrc for other logins.

**Comment by Vogtinator** [comment]
> I was thinking to set 'root-console' instead of virtio for `consoletest_setup` also. Thus, it would set pipefail in root-console and prepare bashrc for other logins.

That would fix `root-console`, but leave `virtio-console` broken. Which is actually worse, as most of the console tests use the latter.

File: schedule/jeos/sle/kvm/jeos-extratest.yaml
Comment (cfconrad) [inline schedule/jeos/sle/kvm/jeos-extratest.yaml:15]: Just curious. We do the reboot here but not for xen-hvm and others, whats the reason for it?

Comment (mloviska) [reply schedule/jeos/sle/kvm/jeos-extratest.yaml:15]: I have just forgotten to update other files. Nevertheless, the apparent problem appears only in test jobs using virtio console.

Comment (cfconrad) [reply schedule/jeos/sle/kvm/jeos-extratest.yaml:15]: thx for explanation!

**Comment by Vogtinator** [comment]
Does it really have to reboot? Wouldn't just 

```
if (has_ttys) {
    select_console('root-console');
    script_run '. /etc/bash.bashrc.local';
}
```

in `consoletest_setup.pm` suffice?

**Review by cfconrad — APPROVED** [review]
LGTM,
I cannot judge, if the reboot after `firstrun` makes something "wrong" to our tests, @Vogtinator WDYT?

**Review by Vogtinator — CHANGES_REQUESTED** [review]
Rebooting makes the tests slower and this also doesn't address the issue on openSUSE.

File: tests/console/consoletest_setup.pm
Comment (Vogtinator) [inline tests/console/consoletest_setup.pm:75]: Needs a comment that this applies to the console switched to by `check_console_font` now

**Review by Vogtinator — APPROVED** [review]
LGTM, assuming VRs for oS and SLE pass. Is there anything with a "special" console situation? That might need a VR as well

**Comment by mloviska** [comment]
> LGTM, assuming VRs for oS and SLE pass. Is there anything with a "special" console situation? That might need a VR as well

I hope we are safe as [has_ttys](
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/Utils/Backends.pm#L110) filters out the problematic backends. I cannot recall any other special thingy atm.

**Pull request merged**
